### ISSUE TITLE
feat: Alert handler should support a Promise

### ISSUE THREAD

**Issue by sean-perkins** [ionic-team/ionic-framework#25700]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already include this feature request, without success.

### Describe the Feature Request

Many of the existing overlays already support a `Promise` to be used with the `handler` to control when the overlay dismisses. This is not available to `ion-alert`.

This feature request is to add support to `ion-alert` to allow for a promise.

### Describe the Use Case

Performing an action in the handler callback that requires asynchronous behavior (performing an API call, waiting for user input, waiting a set duration). 

### Describe Preferred Solution

```ts
handler: async() => {
  const res = await this.someService.method(); // res returns `true` or `false` controlling if the alert dismisses 
  return res;
}
```

### Additional Information

Split from PR #22518

**Comment by sean-perkins** [comment]
This feature is completed and will be available in 6.3.0. This feature request will auto close once it is merged and released. Thanks Hans for the assist!

**Comment by liamdebeasi** [comment]
Closing as it is completed via https://github.com/ionic-team/ionic-framework/pull/25702. Only PRs that get merged into the default branch cause issues to autoclose it seems.
